python/pyspark/ml/linalg/__init__.py

```diff
@@ -101,7 +101,7 @@ def _vector_size(v):
         return len(v)
     elif type(v) in (array.array, list, tuple, xrange):
```

Comment:
If this change is legitimate, we should change this to `isinstance(v, (array.array, list, tuple, xrange))`